participants/qareports.py

```diff
@@ -12,63 +48,127 @@ def handle_wi_control(self, ctrl):
+            self._auth_token = ctrl.config.get("qareports", "auth_token")
             user = ctrl.config.get("qareports", "user")
             password = ctrl.config.get("qareports", "password")
-            realm = ctrl.config.get("qareports", "realm")
-            self.uploader = ReportUploader(
-                apiurl, auth_token, user, password, realm
-            )
+            if user and password:
+                self._auth = (user, password)
+            else:
+                self._auth = None
 
     def handle_wi(self, wid):
 
-        f = wid.fields
-        p = wid.params
-
-        hwproduct = f.qa.hwproduct
-        testtype = f.qa.testtype
-        target = f.qa.target
-        release_version = f.qa.release_version
-        build = f.image.image_url
+        results_dir = wid.params.results_dir
+        if not results_dir:
+            if (
+                wid.fields.test_execution_exit_status and
+                wid.fields.test_execution_exit_status.results_dir
+            ):
+                # Only for Backwards compatibility
+                results_dir = wid.fields.test_execution_exit_status.results_dir
+                self.log.warn(
+                    "Using deprecated field "
+                    "test_execution_exit_status.results_dir"
+                )
+            else:
+                self.log.error("No results_dir defined")
+                return
 
-        if not f.qa or not f.qa.results:
-            self.log.info("no results in this workitem")
+        file_list = list_files(results_dir)
+        if not file_list:
+            self.log.info("No files in results dir %s", results_dir)
             return
-        results = get_results_files_list(f.qa.results.results_dir)
-
-        attachments = []
-        result_xmls = []
-
-        for result in results:
-            if result.endswith(".xml"):
-                result_contents = open(result).read()
-                if result_contents.strip().startswith("<"):
-                    result_xmls.append((result, result_contents))
-                else:
-                    attachments.append((result, result_contents))
-            else:
-                attachments.append((result, open(result).read()))
-
-        if result_xmls:
-            url, msg = self.uploader.send_files(
-                result_xmls,
-                attachments,
-                hwproduct=hwproduct,
-                testtype=testtype,
-                target=target,
-                release_version=release_version,
-                build=build
-            )
 
+        url, msg = self.upload(
+            file_list=file_list,
+            import_params=dict(
+                hwproduct=wid.fields.qa.hwproduct,
+                testtype=wid.fields.qa.testtype,
+                target=wid.fields.qa.target,
+                release_version=wid.fields.qa.release_version,
+                build_txt=wid.fields.image.image_url,
+            ),
+            report_id=wid.params.report_id,
```

Comment:
it seems odd that this is a param given it appears to be used in an interpolation below which suggests that it varies by workitem and not by pdef (unless something is writing dynamic pdefs which is what dister could be nuts enough to do... but I couldn't trivially find that)

Reply:
AFAIK we do not have anything that would update the reports at the moment, so that report_id is not really used. But this leaves it open where to put it if we at some point do. The process can use something like `qareports :report_id => "${f:some_test_thingie.report_id}" ...`